### ISSUE TITLE
Error when creating comparison of imported ships

### ISSUE THREAD

**Issue by WEKarnesky** [cmmcleod/coriolis#149]
Imported ships from a comparison a friend sent me, then created a new comparison and added one of mine plus his ships, got this error message

`Browser:
Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.103 Safari/537.36
Path:
/compare/
Error:
unknown
Message:
dataString.split is not a function
Details:
TypeError: dataString.split is not a function
    at toShip (http://cdn.coriolis.io/app.d7ed0a1f.js:13:1429)
    at r.$scope.addBuild (http://cdn.coriolis.io/app.d7ed0a1f.js:14:1167)
    at fn (eval at <anonymous> (http://cdn.coriolis.io/lib.df42dcb0.js:10:9671), <anonymous>:4:509)
    at e (http://cdn.coriolis.io/lib.df42dcb0.js:10:30681)
    at r.$eval (http://cdn.coriolis.io/lib.df42dcb0.js:8:1044)
    at r.$apply (http://cdn.coriolis.io/lib.df42dcb0.js:8:1274)
    at HTMLTableRowElement.<anonymous> (http://cdn.coriolis.io/lib.df42dcb0.js:10:30733)
    at If (http://cdn.coriolis.io/lib.df42dcb0.js:6:15143)
    at HTMLTableRowElement.Hf.d (http://cdn.coriolis.io/lib.df42dcb0.js:6:15090)`

**Comment by cmmcleod** [comment]
Can you send me the link to the comparison?

**Comment by WEKarnesky** [comment]
http://coriolis.io/comparison/N4IgdiBcIIJgJgTwOQGcAEAlGBJAMugKIwDiehIANCAEZQDaoqUIApgIYDmANq1eCwDKAFwD2AYwDWRLr37iWABgBMhAMzrCARkKq9AYS0B2YwFpTAFguKVAC1MA6HAHctp4gF4PDmK-cwvEABfSiYWDh4%2BaghoOCQ0LFwCYjIKagVoFRg1ABFcnWUYZRzlfWUATgqKxTAAIQtlI2UAVhanP09vfWdagCscTE4ALwBHAA5OD2CAXWoAMwYtLUotRUoxyiNKZso18soANkoLFco1SmVZkAAHFjB2AFsokHgoRSCgA

**Comment by WEKarnesky** [comment]
And this was the ship of mine that I added.

http://coriolis.io/outfit/eagle/22A3A3A1A2A2A2C2929290229272525.AwRj4yg=.Aw1-EA==?bn=Rail%20Fun%20Eagle

**Comment by WEKarnesky** [comment]
One thing I noticed, when in Chrome, the browser I originaly used for this link, I am getting Not a Number (shortened, of course, to NaN) for the prices no matter which ship I go to.  I did a full refresh (using shift or ctrl + f5 to ignore any cached copies) and it still occurs.  If I open a new private browser then things calculate fine (I haven't tested the comparison link in that mode however)

I wonder if closing or restarting will fix it, but left it in case you want me to capture any more data from my client.

**Comment by cmmcleod** [comment]
Hmm. The comparison and build both work fine. 

Can you go to Settings [Top Right] and change the settings for Ship Discount, Component Discount, and Insurance (you can change them back to your original values after you have changed them once). This should hopefully reset the pricing issue

**Comment by WEKarnesky** [comment]
I can confirm that that fixed the pricing.

I'm not sure what cause it, but when I imported the ships from that comparison to my end, it gave the url as:

    http://coriolis.io/outfit/eagle/%5Bobject%20Object%5D?bn=Andy%27s%20RAIL%20EAGLE

**Comment by WEKarnesky** [comment]
correction, I just reopened the comparison to try it and when clicking on the ship it gives the url
 
> http://coriolis.io/outfit/eagle/02A3D3D1E2A2D2C2929290nB4272525.Iw1-EA==.CwBjIRgzq8g=?bn=Andy%27s%20RAIL%20EAGLE

which results in  

    Browser:
Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.103 Safari/537.36
Path:
/outfit/eagle/02A3D3D1E2A2D2C2929290nB4272525.Iw1-EA==.CwBjIRgzq8g=
Error:
unknown
Message:
dataString.split is not a function
Details:
TypeError: dataString.split is not a function
    at toShip (http://cdn.coriolis.io/app.d7ed0a1f.js:13:1429)
    at new <anonymous> (http://cdn.coriolis.io/app.d7ed0a1f.js:21:4581)
    at e (http://cdn.coriolis.io/lib.df42dcb0.js:6:17209)
    at Object.instantiate (http://cdn.coriolis.io/lib.df42dcb0.js:6:17326)
    at http://cdn.coriolis.io/lib.df42dcb0.js:7:6082
    at http://cdn.coriolis.io/lib.df42dcb0.js:12:23333
    at ea (http://cdn.coriolis.io/lib.df42dcb0.js:7:2331)
    at D (http://cdn.coriolis.io/lib.df42dcb0.js:6:28674)
    at g (http://cdn.coriolis.io/lib.df42dcb0.js:6:25046)
    at http://cdn.coriolis.io/lib.df42dcb0.js:6:24687

I'm more of a C# guy, not a webdev guy, but trying the same url in private browsing it works fine.  That seems like there might be a cookie that is corrupting things?

**Comment by cmmcleod** [comment]
Coriolis doesn't use cookies, but local storage. Something has been corrupted with your saved builds.

I'd suggest the following:

Settings -> Back up -> copy and paste the JSON here.

I'll try to import that back up and determine if anything is corrupt

**Comment by WEKarnesky** [comment]
Here you go:

    {
      "builds": {
        "cobra_mk_iii": {
          "Current Ship": "03A4A4A3D3A3C4C1b1b1i1i0m000303B3016a2h.Iw1-kA==.MwBhCYy6co==",
          "Black Betty (Smuggling Rig)": "04D4E4A3E3A3E4C---18000203B33201012f.Iw1-kA==.Aw1-kA=="
        },
        "vulture": {
          "Future Bounty Hunter": "04E5E4E3E5E4E3C1e1o----4e02--2h.Iw19kA==.Aw19kA==",
          "Future Bounty Hunter mk2": "24A5A4A3D5A4C3C1e1e000m0404B233272569.AwRj4zyA.AwiMIyQZgFl4Q===",
          "My Vulture": "24A5A4A3D5A4C3C1e1e000m0404B233272569.AwRj4zyA.CwBjIRgkCYYmBmRFhuEA"
        },
        "cobra_mk_iv": {
          "Explore and mine": "04D4D4D3D3D3D4C2m2m181818--030348v41sC8P42f.Iw18SQ==.Aw1+gLGQ"
        },
        "asp": {
          "Kitsune (Smuggling Rig)": "03A5D5A4D4A5D5C--18---0003-0205390202B40e2f.Iw18WQ==.Aw18WQ=="
        },
        "viper": {
          "Jeremy ship": "03C3C3C2E3A3E2C2727121200-29B42q2h.Iw1+kA==.CwBhrSrUg==="
        },
        "hauler": {
          "Taxi": "02D2D2A1D1D1D2C---30-3u-.Iw19A===.Aw19A==="
        },
        "eagle": {
          "Rail Fun Eagle": "22A3A3A1A2A2A2C2929290229272525.AwRj4yg=.Aw1-EA==",
          "Stock Eagle": {
            "code": "02E3E3E1E2E2E2C1717--44002h-.Iw1-EA==.Aw1-EA==",
            "useName": "Stock Eagle"
          },
          "Andy's RAIL EAGLE": {
            "code": "02A3D3D1E2A2D2C2929290nB4272525.Iw1-EA==.CwBjIRgzq8g=",
            "useName": "Andy's RAIL EAGLE"
          }
        },
        "diamondback": {
          "Jeremy": "24A4A4A2D3A2D4C1g1g1818000m0404B430292h.Iw1-kA==.CwBhGYy7tEg="
        },
        "anaconda": {
          "Proposal 2": "08C7D6D5D8A8D5C1e1e1e1e1b1b--0k02000404040404B005055n3a---v4f42f.Iw18ZlA=.CwBhGYy6dvqA",
          "My Anaconda": "08B7A6A5B8A8C5C2c1e1e1e272719190k02000404040404B005055o2d2d390q6lf42f.Iw18ZlA=.CwBhGYy6dqCYTGVIA==="
        }
      },
      "comparisons": {},
      "insurance": "beta",
      "discounts": [
        1,
        1
      ]
    }

**Comment by cmmcleod** [comment]
Ah. Something caused a bad import (I've seen this before but have not been able to replicate). This will be fixed in version 2.0

Here is the issue:

```
...
    "Stock Eagle": {
        "code": "02E3E3E1E2E2E2C1717--44002h-.Iw1-EA==.Aw1-EA==",
        "useName": "Stock Eagle"
      },
      "Andy's RAIL EAGLE": {
        "code": "02A3D3D1E2A2D2C2929290nB4272525.Iw1-EA==.CwBjIRgzq8g=",
        "useName": "Andy's RAIL EAGLE"
      }
...
```

If you delete the `Stock Eagle` and `Andy's RAIL EAGLE` and then save them again by using these links:

http://coriolis.io/outfit/eagle/02A3D3D1E2A2D2C2929290nB4272525.Iw1-EA==.CwBjIRgzq8g=?bn=Andy%27s%20RAIL%20EAGLE
http://coriolis.io/outfit/eagle/02E3E3E1E2E2E2C1717--44002h-.Iw1-EA==.Aw1-EA==?bn=Stock%20Eagle

This will fix the problem and allow you to continue to using Coriolis.io's current version without further issue

**Comment by WEKarnesky** [comment]
I can confirm that this fixed the issue, thank you.
